nucliadb/nucliadb/search/api/v1/chat.py

```diff
@@ -95,60 +107,68 @@ async def chat_knowledgebox_endpoint(
-    find_request.range_creation_end = item.range_creation_end
-    find_request.range_modification_start = item.range_modification_start
-    find_request.range_modification_end = item.range_modification_end
-    find_request.show = item.show
-    find_request.extracted = item.extracted
-    find_request.shards = item.shards
-    find_request.autofilter = item.autofilter
-    find_request.highlight = item.highlight
-    find_request.with_duplicates = False
-
-    find_results, incomplete = await find(
-        response,
+    chat_request: ChatRequest,
+    user_id: str,
+    client_type: NucliaDBClientType,
+    origin: str,
+    x_synchronous: bool,
+) -> Response:
+    chat_result = await chat(
         kbid,
-        find_request,
-        x_ndb_client,
-        x_nucliadb_user,
-        x_forwarded_for,
-        do_audit=True,
+        chat_request,
+        user_id,
+        client_type,
+        origin,
     )
-    if incomplete:
-        raise IncompleteFindResultsError()
+    if x_synchronous:
+        text_answer = b""
+        async for chunk in chat_result.answer_stream:
+            text_answer += chunk
 
-    return await chat(
-        kbid,
-        user_query,
-        rephrased_query,
-        find_results,
-        item,
-        x_nucliadb_user,
-        x_ndb_client,
-        x_forwarded_for,
-    )
+        relations_results = None
+        if ChatOptions.RELATIONS in chat_request.features:
+            relations_results = await get_relations_results(
+                kbid=kbid, chat_request=chat_request, text_answer=text_answer
+            )
+
+        return Response(
+            content=SyncChatResponse(
+                answer=text_answer.decode(),
+                relations=relations_results,
+                results=chat_result.find_results,
+                status=chat_result.status_code.value,
+            ).json(),
+            headers={
+                "NUCLIA-LEARNING-ID": chat_result.nuclia_learning_id or "unknown",
+                "Access-Control-Expose-Headers": "NUCLIA-LEARNING-ID",
+            },
+        )
+    else:
+
+        async def _streaming_response():
+            bytes_results = base64.b64encode(chat_result.find_results.json().encode())
+            yield len(bytes_results).to_bytes(length=4, byteorder="big", signed=False)
+            yield bytes_results
+
+            text_answer = b""
+            async for chunk in chat_result.answer_stream:
+                text_answer += chunk
+                yield chunk
+
+            yield END_OF_STREAM.encode()
+            if ChatOptions.RELATIONS in chat_request.features:
+                relations_results = await get_relations_results(
+                    kbid=kbid, chat_request=chat_request, text_answer=text_answer
+                )
+                yield base64.b64encode(relations_results.json().encode())
+
+        return StreamingResponse(
+            _streaming_response(),
+            media_type="plain/text",
```

Comment:
the media type is actually `application/octet-stream`, isn't it?

Reply:
I think I just copy/pasted but I'll update nontheless